src/ray/raylet/agent_manager.h

```diff
@@ -89,6 +79,10 @@ class AgentManager : public rpc::AgentManagerServiceHandler {
   virtual void DeleteRuntimeEnvIfPossible(const std::string &serialized_runtime_env,
                                           DeleteRuntimeEnvIfPossibleCallback callback);
 
+  // Wraps this AgentManager to a RuntimeEnvManagerClient. The caller must ensure the
+  // Client lives shorter than this AgentManager.
```

Comment:
Hmm, I think you should have the `AgentManagerAsRuntimeEnvManagerClient` hold a smart pointer to `AgentManager` and lift this restriction on the lifetime. It removes a source of problems and there are no efficiency gains to be had in making it a plain pointer :)

Reply:
ditto, now we are removing agent manager so it is no longer needed.